cmd/clone.go

```diff
@@ -684,13 +690,22 @@ func CloneAllRepos(git git.Gitter, cloneTargets []scm.Repo) {
 		// We use this because we dont want spaces in the final directory, using the web address makes it more file friendly
 		// In the case of root level snippets we use the title which will have spaces in it, the url uses an ID so its not possible to use name from url
 		// With snippets that originate on repos, we use that repo name
-		repoSlug := getAppNameFromURL(repo.URL)
-
+		//
+		// XXX: The URL handling in getAppNameFromURL makes strong presumptions that the URL will end in an
+		// extension like '.git', but this is not the case for sourcehut (and possibly other forges). An scm
+		// can specify its own Slug, but should
+		repoSlug := repo.Slug
+		if repoSlug == "" {
```

Comment:
Lets have the code tell us that this is only a SourceHut issue so we can do like

`if os.Getenv("GHORG_SCM_TYPE") != "sourcehut" {
repoSlug := getAppNameFromURL(repo.URL)
}
`

Reply:
I've wound back all the Repo.Slug stuff and replaced it with the scm type check as requested